File: test/unit/shortest_paths/Edge.cc
// -*- C++ -*-

#include "stlib/shortest_paths/Edge.h"

#include "stlib/shortest_paths/Vertex.h"

#include <iostream>

#include <cassert>

using namespace stlib;

int
main()
{
  using namespace shortest_paths;

  {
    // Default constructor
    Edge< Vertex<int> > edge;
  }
  {
    Vertex<double> source, target;
    Edge< Vertex<double> > edge(&source, &target, 42);
    assert(edge.source() == &source);
    assert(edge.target() == &target);
    assert(edge.weight() == 42);

    std::cout << edge << '\n';
  }
  return 0;
}

File: prototypes/shortest_paths/Edge.py
"The edge class for a graph."

class Edge:
    "An edge in a graph."
    
    def __init__( self, source = None, target = None, weight = None ):
        self.source = source
        self.target = target
        self.weight = weight
        
    def display( self ):
        print "source =", self.source.index, " target =", self.target.index, " weight =", self.weight
